bazel/runfiles.bzl

```diff
@@ -0,0 +1,125 @@
+# Copyright 2022 The Magma Authors.
+
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Inspired by https://github.com/aspect-build/rules_container/blob/main/language/runfiles.bzl by @thesayyn
+# Contributed to https://github.com/aspect-build/rules_container under Apache-2.0
+
+"""
+This file represents a workaround for the current state of https://github.com/bazelbuild/rules_pkg.
+Dependencies are not packaged (even internal dependencies). The rule here expands all
+internal and external dependencies into a PackageFilesInfo so that the files can be used in
+package rules.
+
+The dependencies are not put into service specific paths. This means, that, e.g., pip dependencies and
+proto files that are used by multiple services are only added once in the packaging process.
+
+This rule is currently only applied to python dependencies (go and c/c++ dependencies are linked statically
+into the binaries).
+
+Additionally this rule
+* renames the relative path of files so that they can be found correctly in the target system (usually
+  packaged into the "dist-packages" folder of the used python interpreter)
+* excludes files that are not needed during runtime
+"""
+
+load("@rules_pkg//:providers.bzl", "PackageFilesInfo")
+
+STRIP_PATHS = [
+    "lte/gateway/python/",
+    "orc8r/gateway/python/",
+    "lte/swagger/specs_root/",
+    "orc8r/swagger/specs_root/",
+]
+
+# beware: order matters here, e.g., "lte/protos/oai/" needs to be before "lte/protos/"
+STRIP_PATHS_PROTOS = [
+    "dp/protos/",
+    "feg/protos/",
+    "lte/protos/oai/",
+    "lte/protos/",
+    "orc8r/protos/",
+    "orc8r/swagger/magmad_events_v1",
+]
+
+EXCLUDES = [
+    # external protobuf is only needed during compile time
+    "../com_google_protobuf",
+    # external grpc is only needed during compile time
+    "../com_github_grpc_grpc",
+    "_solib_k8/libexternal_Scom_Ugithub_Ugrpc_Ugrpc_Slibgrpc.so",
```

Comment:
Maybe add a comment here too (whatever that thing is :)).

Reply:
I will. The long description for the record: this is how bazel compiles dynamic libraries - separators like `/` and `.` are encoded -> `libexternal/com.github.grpc.grpc/libgrpc.so`. But it's still a library that is only needed during compile time.

Reply:
addressed